### ISSUE TITLE
Possible cause of safe mode crash?

### ISSUE THREAD

**Issue by BlakeBoxberger** [wizages/It-is-getting-hot-in-here-#1]
I was looking through the source code because I wanted to implement something like this but in the "date" part of the lock screen clock instead, and as I was running through the code on my device by manually calling methods, I think I located an issue with lines 13 and 14:
```
12. City* city = [prefs loadSavedCities][0];

13. [city update];

14. NSArray* hours = city.hourlyForecasts;

```
The city at index 0 for loadSavedCities is null for me. Calling update appears to change nothing and city.hourlyForecasts is empty. Calling hours[0] is what I suspect is ultimately causing the crash.

Thank you for open-sourcing this too! I'll be sure to give you credit if I end up deciding to use/adapt any of this code. (It is open source, right?)

**Comment by wizages** [comment]
Yeah that is where the issue is. I have no idea how I want to fix it. Pretty much you need to call a manual update. I am still working on this.

**Comment by wizages** [comment]
If you have any suggestions or an idea how how to fix it let me know 👍

**Comment by BlakeBoxberger** [comment]
Just learned that opening the Weather app once updates that city. If that helps at all :P

**Comment by wizages** [comment]
Yeah one solution I thought about was manually opening the weather app on respring, but that is sort of ugly. I am trying to dig around and find a better way to do it. The nice thing about doing the opening of the app though is it fixes the problem on all iOS versions.

**Comment by BlakeBoxberger** [comment]
Also it gets reset to null again after every respring. I'm also doing a little digging right now too.

**Comment by BlakeBoxberger** [comment]
You could check if the property on WeatherPreferences is nil. If it's nil, just use the second index.
In WeatherPreferences: `City *localWeatherCity`

**Comment by BlakeBoxberger** [comment]
Also I'm confused about the dependencies, do we need libstatusbar or libstatus9, because I think it's set up as both right now and they conflict.

**Comment by wizages** [comment]
Oh crap havent pushed the newest control file yet one sec!

**Comment by BlakeBoxberger** [comment]
No worries! I think I might have a temporary solution once the control is fixed. I'll submit a pull request if it works :P

**Comment by wizages** [comment]
Should be good to go!

**Comment by BlakeBoxberger** [comment]
JK my solution didn't work... sorry :/ I'll keep trying

**Comment by BlakeBoxberger** [comment]
Oop I think I might have found it. On iOS 10.2, `NSArray* hours = city.hourlyForecasts;` returns an array of WAHourlyForecast objects, which doesn't have a .detail property. Rather, it has a .forecastDetail property which is nil, and I don't see how to update it.

**Comment by BlakeBoxberger** [comment]
A possible solution is using the .temperature property on the WAHourlyForecast object, which returns a WFTemperature object which has properties CGFloat celsius and fahrenheit
